### PR TITLE
Add tests for trait implementations on `strftime::Error`

### PR DESCRIPTION
**Pull request by lopopolo**
- Move `src/tests.rs` to `src/functional_tests.rs` so a `mod tests` can be used in `src/lib.rs`.
- Add test that `impl fmt::Display for Error` does not panic.
- Add test that `impl fmt::Display for Error` does not format to the empty string.
- Add test that `impl std::error::Error for Error` returns inner errors in `std::error::Error::source`.
- Use `Formatter::write_str` instead of `write!` macro in `fmt::Display` implementation. This form is a bit more readable and avoids going through the format machinery.
- Add test for `impl From<TryReserveError> for Error` to ensure `Error::OutOfMemory(..)` variant is returned.

### CODE REVIEW

**Review by x-hgg-x — APPROVED** [review]

**Comment by lopopolo** [comment]
thanks for making those additional fixups.

I've enjoyed getting reviews from you before my changes get merged. How do you feel about both of us requesting review from the other on all PRs?

**Comment by x-hgg-x** [comment]
Yes it is a good idea, we can produce better code in this way.

**Pull request merged**
